### PR TITLE
Give the ability to specify the backend url of the Git Lab Repository

### PR DESCRIPTION
**Pull request by matthieupetite**
Hello, 

We have an in house git lab server and I saw that your extension didn't give the ability to set that URL. So that's the aim on my PR. I also specify that it's the ID of the project that are targeted in the helper message

Cordialement

### CODE REVIEW

File: vss-extension.json [hunk 1]
```diff
@@ -88,6 +88,21 @@
...
+						"inputMode": "textbox",
+						"isConfidential": false,
+						"validation": {
+							"isRequired": true,
+							"dataType": "string"
+						},
+						"values": {
+							"defaultValue": "https://gitlab.com",
+							"inputId": "gitlaburlValues"
+						}
```
Comment (onlyutkarsh) [inline vss-extension.json]: @matthieupetite Thanks for the PR. I am working on this now and I don't think we need a new field to hold this URL. The existing textbox which holds the "Server URL" can be used to connect to gitlab.com or on-premises server URL.
  
image

File: vss-extension.json [hunk 2]
```diff
@@ -181,7 +196,7 @@
 					{
 						"id": "definition",
 						"name": "Source (repository)",
-						"description": "Repository to pull",
+						"description": "ID of the Repository to pull",
```
Comment (onlyutkarsh) [inline vss-extension.json]: This is an editable dropdown and ideally should show existing repos. Why do you think ID needs to be mentioned here?
image

File: vss-extension.json (hunk 1 above)
Comment (matthieupetite) [reply vss-extension.json]: Ok but I use also an In House Team Foundation Server which didn't display that dialog while editing the credential ?

File: vss-extension.json (hunk 2 above)
Comment (matthieupetite) [reply vss-extension.json]: Ok for that my fault ...

File: vss-extension.json (hunk 1 above)
Comment (matthieupetite) [reply vss-extension.json]: image

here is the result on my on prem Team Foundation Server

Comment (onlyutkarsh) [reply vss-extension.json]: > Ok but I use also an In House Team Foundation Server which didn't display that dialog while editing the credential ?

That is really odd. I just tested on latest Azure DevOps server and I see the textbox. What is your TFS version?

image

**Comment by onlyutkarsh** [comment]
@matthieupetite I have fixed this issue in a new release without adding the new field to the service connection UI. Please check and let me know if it solves issues for you. Again appreciate you taking time out to highlight the issue and suggesting the changes.

**Pull request closed without merging**